File: workflow-saga-java/src/main/java/com/acme/workflows/saga/java/Worker.java
package com.acme.workflows.saga.java;

import io.infinitic.workers.InfiniticWorker;

public class Worker {
    public static void main(String[] args) {
        try (InfiniticWorker worker = InfiniticWorker.fromConfigResource("/infinitic.yml", "/saga.yml")) {
            worker.start();
        }
    }
}

File: workflow-saga-kotlin/src/main/kotlin/com/acme/workflows/saga/kotlin/Worker.kt
package com.acme.workflows.saga.kotlin

import io.infinitic.workers.InfiniticWorker

fun main() {
    // infinitic.yml resource is from the dependant "contracts" module
    InfiniticWorker.fromConfigResource("/infinitic.yml", "/saga.yml").use { worker ->
        worker.start()
    }
}
